From 8d8f36400acb03e3308c0887b506009894c0ef3c Mon Sep 17 00:00:00 2001
From: Eric Naim <dnaim@cachyos.org>
Date: Sun, 29 Mar 2026 19:30:17 +0800
Subject: [PATCH 42/51] applesmc: Do not probe for T2 Macbooks

T2 Macbooks must use the applesmc-t2 module because applesmc.c doesn't
provide the necessary features.

Signed-off-by: Eric Naim <dnaim@cachyos.org>
---
 drivers/hwmon/applesmc.c | 3 ++-
 include/linux/cachy-t2.h | 8 ++++++++
 2 files changed, 10 insertions(+), 1 deletion(-)
 create mode 100644 include/linux/cachy-t2.h

diff --git a/drivers/hwmon/applesmc.c b/drivers/hwmon/applesmc.c
index 90a14a7f2..33fbf3022 100644
--- a/drivers/hwmon/applesmc.c
+++ b/drivers/hwmon/applesmc.c
@@ -17,6 +17,7 @@
 
 #define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
 
+#include <linux/cachy-t2.h>
 #include <linux/delay.h>
 #include <linux/platform_device.h>
 #include <linux/input.h>
@@ -1311,7 +1312,7 @@ static int __init applesmc_init(void)
 {
 	int ret;
 
-	if (!dmi_check_system(applesmc_whitelist)) {
+	if (!dmi_check_system(applesmc_whitelist) || apple_is_t2_mac()) {
 		pr_warn("supported laptop not found!\n");
 		ret = -ENODEV;
 		goto out;
diff --git a/include/linux/cachy-t2.h b/include/linux/cachy-t2.h
new file mode 100644
index 000000000..cd73fa5d0
--- /dev/null
+++ b/include/linux/cachy-t2.h
@@ -0,0 +1,8 @@
+#ifndef CACHYOS_T2_H
+#define CACHYOS_T2_H
+
+#include <linux/types.h>
+
+bool apple_is_t2_mac(void);
+
+#endif
-- 
2.55.0

